src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.AliasesAndUsings.cs

```diff
@@ -85,14 +84,41 @@ private SingleNamespaceDeclaration GetMatchingNamespaceDeclaration(CSharpSyntaxN
             throw ExceptionUtilities.Unreachable();
         }
 
-        private static AliasesAndUsings GetOrCreateAliasAndUsings(
+        private AliasesAndUsings GetOrCreateAliasAndUsings(
             ref ImmutableSegmentedDictionary<SingleNamespaceDeclaration, AliasesAndUsings> dictionary,
             SingleNamespaceDeclaration declaration)
         {
-            return RoslynImmutableInterlocked.GetOrAdd(
-                ref dictionary,
-                declaration,
-                static _ => new AliasesAndUsings());
+            if (dictionary.IsDefault)
+            {
+                var builder = ImmutableSegmentedDictionary.CreateBuilder<SingleNamespaceDeclaration, AliasesAndUsings>(ReferenceEqualityComparer.Instance);
```

Comment:
>var builder = ImmutableSegmentedDictionary.CreateBuilder<SingleNamespaceDeclaration, AliasesAndUsings>(ReferenceEqualityComparer.Instance); [](http://example.com/codeflow?start=16&length=139)

This block is essentially the code from the SourceNamespaceSymbol ctor pre [https://github.com/dotnet/roslyn/pull/67655](https://github.com/dotnet/roslyn/pull/67655)